zerver/views/users.py

```diff
@@ -490,3 +490,21 @@ def get_profile_backend(request: HttpRequest, user_profile: UserProfile) -> Http
         result['max_message_id'] = messages[0].id
 
     return json_success(result)
+
+@has_request_variables
+def get_subscription_backend(request: HttpRequest, user_profile: UserProfile,
+                             user_id: int=REQ(validator=check_int, path_only=True),
+                             stream_id: int=REQ(validator=check_int, path_only=True)
+                             ) -> HttpResponse:
+    target_user = access_user_by_id(user_profile, user_id, read_only=True)
+    try:
+        (stream, recipient, sub) = access_stream_by_id(target_user, stream_id)
```

Comment:
We need to call `access_stream_by_id` with `user_profile`, not `target_user`; otherwise we grant any user access to data on an arbitrary stream if they ask whether an admin is subscribed!

Following that, we'll need to do a second query to check the target user's status, which should probably call a dedicated function in `streams.py` that does `Subscription.objects.get(user_profile=target_user, recipient=recipient, is_active=True)` and returns the object if it exists.

(And please add a test for the corner case of trying to access a stream the target user is on but you can't access because you're a guest user or it's a private stream!)